File: is_valid_subsequence/is_valid_subsequence.java
import java.util.List;

public class IsValidSubsequence {
  public static boolean isValidSubsequence(List<Integer> array, List<Integer> sequence) {
    int seqIndex = 0;

    for (int num : array) {
      if (seqIndex == sequence.size()) {
        break;
      }
      if (num == sequence.get(seqIndex)) { 
        seqIndex++;
      }
    }

    return seqIndex == sequence.size();
  }

  public static void main(String[] args) {
    List<Integer> array = List.of(5, 1, 22, 15, 6, -1, 8, 10);
    List<Integer> sequence = List.of(1, 6, -1, 10);
    System.out.println(isValidSubsequence(array, sequence));
  }
}

File: is_valid_subsequence/is_valid_subsequence.ts
function isValidSubsequence(array: number[], sequence: number[]): boolean {
  if (sequence.length > array.length) {
    return false;
  }

  let seqIndex = 0;
  for (const num of array) {
    if (seqIndex === sequence.length) {
      break;
    }
    if (num === sequence[seqIndex]) {
      seqIndex++;
    }
  }

  return seqIndex === sequence.length;
}

const array = [5, 1, 22, 15, 6, -1, 8, 10];
const sequence = [1, 6, -1, 10];
console.log(isValidSubsequence(array, sequence));
